extensions/qute/qute-reference-guide.adoc

```diff
@@ -0,0 +1,673 @@
+----
+<html>
+ <p>Hello {name}! <1>
+</html>
+----
+<1> `{name}` is a value expression that is evaluated when the template is rendered.
+
+Then, we will need to parse the contents into a *template definition* Java object.
+A template definition is an instance of `io.quarkus.qute.Template`.
+
+If using Qute "standalone" you'll need to create an instance of `io.quarkus.qute.Engine` first.
+The `Engine` represents a central point for template management with dedicated configuration. 
+Let's use the convenient builder:
+
+[source,java]
+----
+Engine engine = Engine.builder().addDefaults().build();
+----
+
+TIP: In Quarkus, there is a preconfigured `Engine` available for injection - see <<quarkus_integration>>.
+
+If we have an `Engine` instance we could parse the template contents:
+
+[source,java]
+----
+Template helloTemplate = engine.parse(helloHtmlContent);
+----
+
+TIP: In Quarkus, you can simply inject the template definition. The template is automatically parsed and cached - see <<quarkus_integration>>.
+
+Finally, we will create a *template instance*, set the data and render the output:
+
+[source,java]
+----
+// Renders <html><p>Hello Jim!</p></html>
+helloTemplate.data("name", "Jim").render(); <1> 
+----
+<1> `Template.data(String, Object)` is a convenient method that creates a template instance and sets the data in one step.
+
+So the workflow is simple: 
+
+1. Create template contents (`hello.html`),
+2. Parse template definition (`io.quarkus.qute.Template`),
+3. Create template instance (`io.quarkus.qute.TemplateInstance`),
+4. Render output.
+
+The `Engine` is able to cache the definitions so that it's not necessary to parse the contents again and again.
+
+NOTE: In Quarkus, the caching is done automatically.
+
+== Core Features
+
+=== Syntax and Building Blocks
+
+The dynamic parts of a template include:
+
+* *comment* 
+** `{! This is a comment !}`,
+** could be multi-line,
+** may contain expressions and sections: `{! {#if true} !}`.
+* *expression*
+** outputs the evaluated value,
+** simple properties: `{foo}`, `{item.name}`,
+** virtual methods: `{item.get(name)}`, `{name ?: 'John'}`,
+** with namespace: `{inject:colors}`.
+* *section tag*
+** may contain expressions and sections: `{#if foo}{foo.name}{/if}`,
+** the name in the closing tag is optional: `{#if active}ACTIVE!{/}`,
+** can be empty: `{#myTag image=true /}`,
+** may declare nested section blocks: `{#if item.valid} Valid. {#else} Invalid. {/if}` and decide which block to render.
+
+==== Expressions
+
+An expression consists of:
+
+* an optional namespace followed by a colon `:`, 
+* parts separated by dot `.`.
+
+The first part of the expression is always resolved against the <<current_context_object, current context object>>.
+If no result is found for the first part it's resolved against the parent context object (if available).
+For an expression that starts with a namespace the current context object is found using the available `NamespaceResolver` s.
```

Comment:
You can use double backticks when you're in this case. Then you don't need a space after.